src/Components/APICalls.js

```diff
@@ -4,15 +4,24 @@ const fiveMinutesMs = 1000 * 60 * 5;
 
 const apiUrl = `https://api-jet-lfoguxrv7q-uw.a.run.app`;
 
-export async function APISearch(inputValue) {
-  let response = await fetch(`${apiUrl}/anime/search?query=${inputValue}`, {
-    mode: "cors",
-  });
-  await handleErrors(response);
-  let responseJson = await response.json();
-  let temp = [];
-  responseJson.items.map((item, index) => temp.push(item));
-  return temp;
+export function useAPISearch(inputValue) {
+  return useQuery(
+    ["search:" + inputValue],
+    async () => {
+      let response = await fetch(`${apiUrl}/anime/search?query=${inputValue}`, {
+        mode: "cors",
+      });
+      await handleErrors(response);
+      let responseJson = await response.json();
+      let temp = [];
```

Comment:
If the point of this code with `temp` is to return an array of all the items in `responseJson.items`, could this all be replaced with the following code?
```js
if (inputValue.length === 0) {
  return [];
}
return responseJson.items;
```

Reply:
Squeaky clean - very nice!!